core/src/main/java/org/apache/calcite/materialize/MaterializationService.java

```diff
@@ -86,16 +86,17 @@ private MaterializationService() {
   /** Defines a new materialization. Returns its key. */
   public MaterializationKey defineMaterialization(final CalciteSchema schema,
       TileKey tileKey, String viewSql, List<String> viewSchemaPath,
-      final String suggestedTableName, boolean create, boolean existing) {
+      final String suggestedTableName,  List<String> suggestedTableSchemaPath,
+      boolean create, boolean existing) {
     return defineMaterialization(schema, tileKey, viewSql, viewSchemaPath,
-        suggestedTableName, tableFactory, create, existing);
+        suggestedTableName, suggestedTableSchemaPath, tableFactory, create, existing);
   }
 
   /** Defines a new materialization. Returns its key. */
   public MaterializationKey defineMaterialization(final CalciteSchema schema,
       TileKey tileKey, String viewSql, List<String> viewSchemaPath,
-      String suggestedTableName, TableFactory tableFactory, boolean create,
-      boolean existing) {
+      String suggestedTableName, List<String> suggestedTableSchemaPath,
```

Comment:
I see what you mean about these suggested properties - they don't really fit our use case as we should never really be generating a materialization inside this service.